test.py

```diff
@@ -249,6 +254,21 @@ def test_gossip(node_factory, bitcoind, impls):
     wait_for(lambda: len(node2.getnodes()) == 5, interval=1)
 
 
+@pytest.mark.parametrize("impls", product(impls, repeat=2), ids=idfn)
```

Comment:
you dont need a pair of nodes to run this test, just use 
` @pytest.mark.parametrize("impl", impls, ids=idfn)`

Reply:
changed, thanks!